sdk/storage/blobs/blob_client.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+namespace azure { namespace storage { namespace blobs {
+
+class blob_client
+{
+public:
+    // connection string
+    blob_client(const std::string& connection_string, std::string container_name, std::string blob_name, blob_client_options options);
+
+    // shared key auth 
+    blob_client(azure::core::http::uri uri, azure::storage::common::storage_shared_key_credential credential, blob_client_options options);
+
+    // token auth
+    blob_client(azure::core::http::uri uri, azure::core::token_credential credential, blob_client_options options);
+
+    // anonymous/SAS/customized pipeline auth
+    blob_client(azure::core::http::uri uri, blob_client_options options);
+
+    bool is_snapshot() const;
+
+    azure::core::http::response<void> delete_(const delete_blob_options& options = delete_blob_options());
+
+    azure::core::http::response<bool> exists(const blob_exists_options& options = blob_exists_options());
```

Comment:
Mark: Response returns get blob propertie's response, will it be OK?

Reply:
Then there's no difference between this API and `get_properties`, why don't we remove this API?

Reply:
Well, the behavior is a bit different. `get_properties` will throw exception when the blob does not exist while this API returns false.

Reply:
I cannot agree because

1. It doesn't align with C#.

2. The semantic of `blob_properties` is incongruous with API name `exists`.

3. It's hard to use, for example, if returning `bool`, we can use it like this (assuming response.value() returns the template value)

```C++
if (blob_client.exists().value())
{
    std::cout << "Blob exists\n"
}
```

otherwise, we have to use it like this

```C++
if (!blob_client.exists().value().empty())
{
    std::cout << "Blob exists\n";
}
```